### PR TITLE
[Iceberg]Support setting the max number of columns for which metrics are collected

### PR DESCRIPTION
**Pull request by hantangwangd**
## Description

For a very-wide-table, there will be too much metadata stored as stats in manifest files if we always collect stats for all the columns. It will occupy too much memory when loading the table metadata into memory. To cope with this, Iceberg lib support configuring the max columns that can have metrics for a table, referring to: https://github.com/apache/iceberg/pull/3959, https://github.com/apache/iceberg/pull/5215, https://github.com/apache/iceberg/pull/5916

This PR add an Iceberg table property `metrics_max_inferred_column` to set the max columns number for which metrics are collected, and implement recognizing and supporting `metrics_max_inferred_column` on metrics collection for Iceberg table with `PARQUET` format.

## Motivation and Context

Support setting the max number of columns for which metrics are collected

## Impact

Iceberg tables with `PARQUET` format now only collect statistics information from the first 100 columns by default

## Test Plan

 - Newly added test case to show the behavior of stats collection with columns count limit

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

Iceberg Connector Changes
* Add table property `metrics_max_inferred_column` to configure the max columns number for which metrics are collected, and support `metrics_max_inferred_column` for Iceberg table with `PARQUET` format :pr:`23468`
```

### CODE REVIEW

**Review by steveburnett — APPROVED** [review]
LGTM! (docs) 

Pull branch, local doc build, looks good. Thanks!

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSink.java [hunk 1]
```diff
@@ -105,10 +106,10 @@ public class IcebergPageSink
     private long writtenBytes;
     private long systemMemoryUsage;
     private long validationCpuNanos;
+    private Table table;
 
     public IcebergPageSink(
-            Schema outputSchema,
-            PartitionSpec partitionSpec,
+            Table table,
```
Comment (tdcmeehan) [inline presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSink.java:112]: Can you help explain why we need these changes to `IcebergPageSink`, and why do we need to pass in the Iceberg `Table` when it seems we're just using the Presto `PrestoIcebergTable` implementation?

Comment (hantangwangd) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSink.java:112]: The problem encountered when creating `IcebergFileWriter` in `IcebergPageSink` is that, among all the static methods provided by Iceberg Lib for building a `MetricsConfig` instance, we can only pass an iceberg `Table` object in order to specify the table's `schema`. Refer to the static methods in `MetricsConfig`.

However, at present, there isn't a way on the presto worker side for us to load an iceberg `Table` object through metadata manager. So we just built an iceberg `Table` interface's implementation that is only used for simple encapsulation of `schema`, `properties`, 'spec', and 'sortOrder'. Then pass to and use it in `IcebergPageSink`.

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java [hunk 2]
```diff
@@ -222,6 +225,14 @@ public static Table getIcebergTable(ConnectorMetadata metadata, ConnectorSession
         return icebergMetadata.getIcebergTable(session, table);
     }
 
+    public static Table getShallowWrappedIcebergTable(Schema schema, PartitionSpec spec, SortOrder sortOrder, Map<String, String> properties)
```
Comment (tdcmeehan) [inline presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java:228]:
```suggestion
    public static Table getShallowWrappedIcebergTable(Schema schema, PartitionSpec spec, Map<String, String> properties, Optional<SortOrder> sortOrder)
```

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/PrestoIcebergTable.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+import org.apache.iceberg.UpdateProperties;
+import org.apache.iceberg.UpdateSchema;
+import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.LocationProvider;
+
+import java.util.List;
+import java.util.Map;
+
+public class PrestoIcebergTable
```
Comment (tdcmeehan) [inline presto-iceberg/src/main/java/com/facebook/presto/iceberg/PrestoIcebergTable.java:47]:
```suggestion
public class PrestoIcebergTableForMetricsConfig
```

Please add a comment that this is a dummy class and is required for `MetricsConfig#forTable`.  Consider an Iceberg issue and PR adding a more direct method.

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSinkProvider.java [hunk 4]
```diff
@@ -80,9 +82,9 @@ private ConnectorPageSink createPageSink(ConnectorSession session, IcebergWritab
         PartitionSpec partitionSpec = toIcebergPartitionSpec(tableHandle.getPartitionSpec()).toUnbound().bind(schema);
         LocationProvider locationProvider = getLocationProvider(new SchemaTableName(tableHandle.getSchemaName(), tableHandle.getTableName().getTableName()),
                 tableHandle.getOutputPath(), tableHandle.getStorageProperties());
+        Table table = getShallowWrappedIcebergTable(schema, partitionSpec, null, tableHandle.getStorageProperties());
```
Comment (tdcmeehan) [inline presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSinkProvider.java:85]:
```suggestion
        Table table = getShallowWrappedIcebergTable(schema, partitionSpec, tableHandle.getStorageProperties(), Optional.empty());
```

**Review by tdcmeehan — CHANGES_REQUESTED** [review]

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/PrestoIcebergTable.java (hunk 3 above)
Comment (hantangwangd) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/PrestoIcebergTable.java]: > Please add a comment that this is a dummy class and is required for MetricsConfig#forTable.

The comment is added, please take a look when available.

> Consider an Iceberg issue and PR adding a more direct method.

Thanks for the suggestion, will consult with the Iceberg community later on whether a more direct method should be added.

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/PrestoIcebergTableForMetricsConfig.java [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.iceberg.SortOrder.unsorted;
+
+/**
+ * This is a dummy class required for {@link org.apache.iceberg.MetricsConfig#forTable}
+ *
+ * */
```
Comment (tdcmeehan) [inline presto-iceberg/src/main/java/com/facebook/presto/iceberg/PrestoIcebergTableForMetricsConfig.java:54]:
```suggestion
/**
 * This is a dummy class required for {@link org.apache.iceberg.MetricsConfig#forTable}
 **/
```

**Review by tdcmeehan — DISMISSED** [review]

**Review by tdcmeehan — APPROVED** [review]

**Review by ZacBlanco — APPROVED** [review]

**Pull request merged**